code/game/objects/items/weapons/tools.dm

```diff
@@ -399,6 +399,9 @@
 		playsound(src.loc, 'sound/effects/refill.ogg', 50, 1, -6)
 		return
 	else if(istype(O, /obj/structure/reagent_dispensers/fueltank) && O.Adjacent(user) && welding)
+		if(isrevenant(user))
+			to_chat(user, SPAN_WARNING("A strange force prevents you from doing this.")) //there is no way to justify this icly
+			return
```

Comment:
This feels hardcoded and icky. I'm not a fan of it. Maybe tie it to advancedtooluser or add a new proc that controls these things.